internal/executor/instance/instance.go

```diff
@@ -98,14 +116,23 @@ func RunDockerizedAgent(ctx context.Context, config *RunConfig, params *Params)
 		return err
 	}
 
-	logger.WithContext(ctx).Debugf("pulling image %s", params.Image)
-	progress, err := cli.ImagePull(ctx, params.Image, types.ImagePullOptions{})
-	if err != nil {
-		return err
+	var noPull bool
+
+	// Handle prebuilt instance images being available only locally
+	if strings.HasPrefix(params.Image, "gcr.io/cirrus-ci-community/") {
+		noPull = true
 	}
-	_, err = io.Copy(ioutil.Discard, progress)
-	if err != nil {
-		return err
+
+	if !noPull {
+		logger.WithContext(ctx).Debugf("pulling image %s", params.Image)
+		progress, err := cli.ImagePull(ctx, params.Image, types.ImagePullOptions{})
```

Comment:
Maybe we should just check local cache if the image exists and pull only if not? Might be useful for folks to check their custom build environments locally. I some use cases where people crafted an image for CI manually or locally and just pushed it to Docker hub. With the current pull there is no way for them to test new images without pushing them. 🤔

Reply:
>Maybe we should just check local cache if the image exists and pull only if not?

>With the current pull there is no way for them to test new images without pushing them.

This is a valid use-case but note that this would prevent the image from ever being updated, which makes some things harder to diagnose when working with rapidly-changing images.

The other option is to always pull the image but never bail-out due to a failed pull, just simply log the warning message and continue.

For now the solution you've proposed indeed seems to be adequate, but in the future we'd probably have to add a new command-line argument for enabling the pulls since a configuration file may contain a lot of images, and manually updating them would be too tedious.

Reply:
`docker run` does use a local image if available instead of always trying to pull it. Behaviour I described is closer to this. We can think of having a `cirrus pull` command that can re-pull images needed by tasks. 🤔

Reply:
See https://github.com/cirruslabs/cirrus-cli/pull/33/commits/92e0fc87b200f32626ba1ef0ac9b40388f4d9c56.